### PR TITLE
Update Jenkins repository key URLs.

### PR DESCRIPTION
**Pull request by nuclearsandwich**
# Description

The repository signing keys for the Jenkins apt and rpm repositories have been rotated and the new signing keys are available at a separate URL from the previous ones.

As of 2023-06-19 the previous keys are no longer available and the URLs used prior to this patch 404 so I do not think there is an advantage to making this change conditional.

Background on the Jenkins repository change is here: https://www.jenkins.io/blog/2023/03/27/repository-signing-keys-changing/

## Check List

- [x] A summary of changes made is included in the CHANGELOG under `## Unreleased`
- ~~New functionality includes testing.~~
- ~~New functionality has been documented in the README if applicable.~~

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@ This file is used to list changes made in each version of the jenkins cookbook.
 
 ## Unreleased
 
+Update Jenkins apt/rpm repository key urls.
```
Comment (nuclearsandwich) [inline CHANGELOG.md:7]: The most recent changelog entries do not use bulleted lists. I followed that although earlier entries do. Was there an explicit style change or was this a formatting discrepancy with the recent entries?

**Comment by nuclearsandwich** [comment]
CI blocked re-running integration tests because of broken links to tickets.chef.io in the changelog (although these links were added in much earlier entries).

I don't think that it makes sense to disable link checking completely or to allow the markdown check to fail every time the changelog is touched. Can we ignore those specific urls somehow?

It seems like the markdown-link-check supports a json config file but I don't know if adding one here would be properly plumbed through the lint-unit workflow without making changes there. https://github.com/sous-chefs/.github/blob/main/.github/workflows/lint-unit.yml

**Comment by ramereth** [comment]
@nuclearsandwich broken links should be fixed or just removed. I have been removing those links in other CHANGELOG's too and I think the same should be the case here.

File: CHANGELOG.md (hunk 1 above)
Comment (ramereth) [reply CHANGELOG.md:7]: Those were done via automated PRs which don't know about the bullets. Please add a bullet here and try and fix the formatting of the others if you can.

**Review by ramereth — CHANGES_REQUESTED** [review]

**Review by xorima — DISMISSED** [review]

**Comment by kitchen-porter** [comment]
Released as: [9.5.19](https://github.com/sous-chefs/jenkins/releases/tag/9.5.19)

**Pull request merged**